packages/ra-data-graphql/README.md

```diff
@@ -0,0 +1,230 @@
+                ...
+            ]
+        },
+        ...
+    ],
+    queries: [
+        {
+            name: 'createPost',
+            args: [
+                { name: 'title', type: { kind: 'NON_NULL', ofType: { kind: 'SCALAR', name: 'String' } } }
+            ],
+            type : { kind: 'OBJECT', name: 'Category' }
+        },
+        ...
+    ],
+    resources: [
+        {
+            type: {
+                name: 'Post',
+                kind: 'OBJECT',
+                fields: [
+                    { name: 'id', type: { kind: 'NON_NULL', ofType: { kind: 'SCALAR', name: 'ID' } } },
+                    { name: 'title', type: { kind: 'NON_NULL', ofType: { kind: 'SCALAR', name: 'String' } } },
+                    ...
+                ]
+            },
+            GET_LIST: {
+                name: 'createPost',
+                args: [
+                    { name: 'title', type: { kind: 'NON_NULL', ofType: { kind: 'SCALAR', name: 'String' } } }
+                ],
+                type : { kind: 'OBJECT', name: 'Category' }
+            },
+            ...
+        }
+    ]
+}
+```
+
+The `queryBuilder` function must return a function which will be called with the same parameters as the react-admin client but must return an object matching the `options` of the ApolloClient [query](http://dev.apollodata.com/core/apollo-client-api.html#ApolloClient.query) method with an additional `parseResponse` function.
+
+This `parseResponse` function will be called with an [ApolloQueryResult](http://dev.apollodata.com/core/apollo-client-api.html#ApolloQueryResult) and must returns the data expected by react-admin.
+
+For example:
+
+```js
+import buildFieldList from './buildFieldList';
+
+const queryBuilder = introspectionResults => (raFetchType, resourceName, params) => {
+    const resource = introspectionResults.resource.find(r => r.type.name === resourceName);
+
+    switch (raFetchType) {
+        case 'GET_ONE':
+            return {
+                query: gql`query ${resource[raFetchType].name}($id: ID) {
+                    data: ${resource[raFetchType].name}(id: $id) {
+                        ${buildFieldList(introspectionResults, resource, raFetchType)}
+                    }
+                }`,
+                variables: params, // params = { id: ... }
+                parseResponse: response => response.data,
+            }
+            break;
+        // ... other types handled here
+    }
+}
+```
+
+```js
+buildGraphQLProvider({ queryBuilder });
+```
+
+## Troubleshooting
+
+*When I create or edit a resource, the list or edit page does not refresh its data*
+
+`react-admin` maintain its own cache of resources data but, by default, so does the Apollo client. For every queries, we inject a default [`fetchPolicy`](http://dev.apollodata.com/react/api-queries.html#graphql-config-options-fetchPolicy) set to `network-only` so that the Apollo client always refetch the data when requested.
```

Comment:
data provider

Reply:
Not this time: we are talking about the Apollo client, not the GraphQL dataProvider which uses this client